R/addRasterRGB.R

```diff
@@ -0,0 +1,194 @@
+#' different bandwidths in the electromagnetic spectrum) are combined such
+#' that they represent the red, green and blue channel. This function can
+#' be used to make 'true (or false) color images' from Landsat and other
+#' multi-band satellite images. Note, this text is plagirized, i.e. copied
+#' from \code{\link{plotRGB}}.
+#' \code{AddRasterRGB} and \code{addStarsRGB} are aliases.
+#'
+#' @param map a map widget object created from `leaflet()``
+#' @param x a  `RasterBrick`, `RasterStack` or `stars`` raster object
+#' @param r integer. Index of the Red channel/band, between 1 and nlayers(x)
+#' @param g integer. Index of the Green channel/band, between 1 and nlayers(x)
+#' @param b integer. Index of the Blue channel/band, between 1 and nlayers(x)
+#' @param quantiles the upper and lower quantiles used for color stretching.
+#' If set to NULL, no stretching is applied.
+#' @param maxpixels integer > 0. Maximum number of cells to use for the plot.
+#' If maxpixels < \code{ncell(x)}, sampleRegular is used before plotting.
+#' @param na.color the color to be used for NA pixels
+#' @param method Method used to compute
+#' values for the resampled layer that is passed on to leaflet. mapview does
+#' projection on-the-fly to ensure correct display and therefore needs to know
+#' how to do this projection. The default is 'bilinear' (bilinear interpolation),
+#' which is appropriate for continuous variables. The other option, 'ngb'
+#' (nearest neighbor), is useful for categorical variables.
+#' @param ... additional arguments passed on to \code{\link{addRasterImage}}
+#'
+#' @author
+#' Tim Appelhans, Luigi Ranghetti
+#'
+#' @examples
+#' if (interactive()) {
+#'   library(raster)
+#'   library(stars)
+#'   library(plainview)
+#'   library(leaflet)
+#'
+#'   leaflet() %>%
+#'     addTiles(group = "OpenStreetMap") %>%
+#'     addRasterRGB(plainview::poppendorf, 4,3,2, group = "True colours") %>%
+#'     addStarsRGB(st_as_stars(plainview::poppendorf), 5,4,3, group = "False colours") %>%
+#'     addLayersControl(
+#'       baseGroups = c("Satellite"),
+#'       overlayGroups = c("True colours", "False colours"),
+#'     )
+#' }
+#'
+#' @importFrom grDevices rgb
+#' @importFrom leaflet addRasterImage
+#' @importFrom raster extent as.factor is.factor ncell projectExtent
+#'  projectRaster projection sampleRegular
+#' @importFrom stats quantile
+#' @importFrom sp CRS
+#' @importFrom scales rescale
+#' @export
+
+addRasterRGB <- function(map, x, r = 3, g = 2, b = 1,
+                         quantiles = c(0.02, 0.98),
+                         maxpixels = 5e+05,
+                         na.color = "#BEBEBE80",
+                         method = c("bilinear", "ngb"),
+                         ...) {
+
+  if (inherits(map, "mapview")) map = mapview2leaflet(map)
+  method = match.arg(method)
+
+  if (inherits(x, "Raster")) {
+
+    x <- rasterCheckSize(x, maxpixels)
+    xout <- CheckAdjustProjection(x, method)
```

Comment:
I removed these lines (the CRS conversion which was done in `CheckAdjustProjection()` is already done in `addRasterImage` / `addStarsImage`, while the raster resize which was done in `rasterCheckSize()` is not yet done).
I removed the arguments `maxpixels` and `method`, leaving `na.color` since this is an information which can not be passed to `addRasterImage` / `addStarsImage`.